dashboard/test/ui/features/step_definitions/steps.rb

```diff
@@ -1032,6 +1048,34 @@ def set_cookie(key, value)
   @browser.manage.add_cookie params
 end
 
+def set_cookie_dcdo(key, value)
+  cookie_dcdo =
+    begin
+      JSON.parse(@browser.manage.cookie_named('DCDO').try(:[], :value).presence || '{}')
+    rescue Selenium::WebDriver::Error::NoSuchCookieError
+      {}
+    end
+
+  cookie_dcdo[key] = value
+
+  begin
+    set_cookie('DCDO', cookie_dcdo.to_json)
+  rescue Selenium::WebDriver::Error::InvalidCookieDomainError
+    navigate_to replace_hostname('http://studio.code.org')
```

Comment:
The error occurs when we try to set a cookie for a domain while being on a different domain, which is not allowed. To avoid this, we should be within the expected domain scope `.code.org`:
https://github.com/code-dot-org/code-dot-org/blob/f8b90d55464aa91daa5dfaba200888d7ad634de3/dashboard/test/ui/features/step_definitions/steps.rb#L1043-L1046

Reply:
I believe Pegasus shares the same `.code.org` domain with Studio, and I wasn't aware of `hourofcode.com`. However, I can provide support for any domain if needed.

Reply:
I believe Pegasus shares the same `.code.org` domain with Studio, and I wasn't aware of `hourofcode.com`.
However, after considering this, I realized it would be better to remove the rescue for `InvalidCookieDomainError`.